src/hexkit/protocols/dao.py

```diff
@@ -38,7 +39,7 @@
     "MultipleHitsFoundError",
     "Dao",
     "DaoFactoryProtocol",
-    "BaseModelWithId",
+    "UUID4Field",
```

Comment:
Maybe also import and re-export UUID4 for convenience, so that both can be imported in the same statement.

Reply:
In most cases where you would need `UUID4Field`, you're defining a model that subclasses `BaseModel`. If you're importing `BaseModel`, you can also import `UUID4`, so it probably doesn't offer much of an advantage to import-export it here.

Reply:
Agreed.